Les sous-répertoires /usr/local/insa/GNAT-2021-dnu/2021/arm-elf/arm-eabi/lib/gnat/ravenscar-full-stm32f746disco/adalib et ./adalib sont identiques
Seulement dans ./: bsp
Seulement dans ./: bsp_build.gpr
Seulement dans ./: .clang-format
Seulement dans ./: debug_cubeide
Seulement dans ./: default.cgpr
Seulement dans ./: .gitignore
Les sous-répertoires /usr/local/insa/GNAT-2021-dnu/2021/arm-elf/arm-eabi/lib/gnat/ravenscar-full-stm32f746disco/gnarl et ./gnarl sont identiques
Seulement dans ./: gnarl-2020
Seulement dans ./: gnarl_empty
Les sous-répertoires /usr/local/insa/GNAT-2021-dnu/2021/arm-elf/arm-eabi/lib/gnat/ravenscar-full-stm32f746disco/gnarl_user et ./gnarl_user sont identiques
Les sous-répertoires /usr/local/insa/GNAT-2021-dnu/2021/arm-elf/arm-eabi/lib/gnat/ravenscar-full-stm32f746disco/gnat et ./gnat sont identiques
Seulement dans ./: gnat-2020
Les sous-répertoires /usr/local/insa/GNAT-2021-dnu/2021/arm-elf/arm-eabi/lib/gnat/ravenscar-full-stm32f746disco/gnat_user et ./gnat_user sont identiques
Seulement dans ./: gprconfig-db
Les sous-répertoires /usr/local/insa/GNAT-2021-dnu/2021/arm-elf/arm-eabi/lib/gnat/ravenscar-full-stm32f746disco/ld et ./ld sont identiques
Les sous-répertoires /usr/local/insa/GNAT-2021-dnu/2021/arm-elf/arm-eabi/lib/gnat/ravenscar-full-stm32f746disco/ld_user et ./ld_user sont identiques
Les sous-répertoires /usr/local/insa/GNAT-2021-dnu/2021/arm-elf/arm-eabi/lib/gnat/ravenscar-full-stm32f746disco/obj et ./obj sont identiques
diff -c /usr/local/insa/GNAT-2021-dnu/2021/arm-elf/arm-eabi/lib/gnat/ravenscar-full-stm32f746disco/ravenscar_build.gpr ./ravenscar_build.gpr
*** /usr/local/insa/GNAT-2021-dnu/2021/arm-elf/arm-eabi/lib/gnat/ravenscar-full-stm32f746disco/ravenscar_build.gpr	2025-04-03 17:37:35.866929122 +0200
--- ./ravenscar_build.gpr	2021-11-17 09:57:08.000000000 +0100
***************
*** 1,34 ****
- with "target_options.gpr";
  with "runtime_build.gpr";
  
  project Ravenscar_Build is
!    for Languages use ("Ada", "Asm_Cpp");
  
     for Runtime ("Ada") use Runtime_Build'Runtime("Ada");
     for Target use Runtime_Build'Target;
- 
     for Library_Auto_Init use "False";
     for Library_Name use "gnarl";
     for Library_Kind use "static";
- 
     for Library_Dir use "adalib";
     for Object_Dir use "obj";
  
!    for Source_Dirs use ("gnarl_user", "gnarl");
! 
!    package Compiler is
!       for Default_Switches ("C") use Target_Options.GNARL_CFLAGS;
!       for Default_Switches ("Ada") use Target_Options.GNARL_ADAFLAGS;
!       for Default_Switches ("Asm_Cpp") use Target_Options.ASMFLAGS;
  
!       --  Always build System.BB.CPU_Primitives at -O2 as some targets may make
!       --  use of inline assembly that rely on specific compiler behaviour to
!       --  load constants that are not enabled at -O0.
!       for Switches ("s-bbcppr.adb") use Target_Options.GNARL_ADAFLAGS & ("-O2");
!       --  Some runtime files need to be compiled with debug info, so that gdb
!       --  is not blind.
!       for Switches ("s-tasdeb.adb") use Target_Options.ALL_ADAFLAGS
!         & ("-g", "-O0");
!    end Compiler;
  
! end Ravenscar_Build;
\ Pas de fin de ligne à la fin du fichier
--- 1,18 ----
  with "runtime_build.gpr";
  
  project Ravenscar_Build is
!    for Languages use ("C");
  
     for Runtime ("Ada") use Runtime_Build'Runtime("Ada");
     for Target use Runtime_Build'Target;
     for Library_Auto_Init use "False";
     for Library_Name use "gnarl";
     for Library_Kind use "static";
     for Library_Dir use "adalib";
     for Object_Dir use "obj";
  
!    for Source_Dirs use ("gnarl_empty");
  
!    package Compiler renames Runtime_Build.Compiler;
  
! end Ravenscar_Build;
Seulement dans /usr/local/insa/GNAT-2021-dnu/2021/arm-elf/arm-eabi/lib/gnat/ravenscar-full-stm32f746disco/: ravenscar_build_seb.gpr
Seulement dans /usr/local/insa/GNAT-2021-dnu/2021/arm-elf/arm-eabi/lib/gnat/ravenscar-full-stm32f746disco/: README
diff -c /usr/local/insa/GNAT-2021-dnu/2021/arm-elf/arm-eabi/lib/gnat/ravenscar-full-stm32f746disco/runtime_build.gpr ./runtime_build.gpr
*** /usr/local/insa/GNAT-2021-dnu/2021/arm-elf/arm-eabi/lib/gnat/ravenscar-full-stm32f746disco/runtime_build.gpr	2025-06-18 10:55:53.381650993 +0200
--- ./runtime_build.gpr	2025-07-02 13:53:49.874399061 +0200
***************
*** 1,56 ****
  with "target_options.gpr";
  
  project Runtime_Build is
!   for Languages use ("Ada", "C", "Asm_Cpp");
!   for Runtime ("Ada") use Project'Project_Dir;
!   --for Target use "arm-eabi";
!   for Target use "arm-linux-gnueabi";
! 
!   for Library_Auto_Init use "False";
!   for Library_Name use "gnat";
!   for Library_Kind use "static";
! 
!   for Library_Dir use "adalib";
!   for Object_Dir use "obj";
! 
!   for Source_Dirs use ("gnat_user", "gnat");
! 
!   package Naming is
!      for Spec_Suffix ("Asm_CPP") use ".inc";
!   end Naming;
! 
!   package Compiler is
!      for Default_Switches ("C") use Target_Options.ALL_CFLAGS;
!      for Default_Switches ("Ada") use Target_Options.ALL_ADAFLAGS;
!      for Default_Switches ("Asm_Cpp") use Target_Options.ASMFLAGS;
! 
!      --  Some runtime files need to be compiled with debug info, so that gdb
!      --  is not blind.
!      for Switches ("s-traceb.adb") use Target_Options.ALL_ADAFLAGS
!        & ("-g")
!        & ("-fno-optimize-sibling-calls", "-fno-inline-functions-called-once");
!      for Switches ("a-except.adb") use Target_Options.ALL_ADAFLAGS
!        & ("-g", "-O1", "-fno-inline", "-fno-toplevel-reorder");
!      for Switches ("s-excdeb.adb") use Target_Options.ALL_ADAFLAGS
!        & ("-g", "-O0");
!      for Switches ("s-assert.adb") use Target_Options.ALL_ADAFLAGS
!        & ("-g");
!      for Switches ("a-tags.adb") use Target_Options.ALL_ADAFLAGS
!        & ("-g");
!      for Switches ("raise-gcc.c") use Target_Options.ALL_CFLAGS
!        & ("-fexceptions");
! 
!      --  Don't inline System.Machine_Reset otherwise we can loose our common
!      --  exit system.
! 
!      for Switches ("s-macres.adb") use Target_Options.ALL_ADAFLAGS
!        & ("-fno-inline");
! 
!      --  Generate ada_target_properties to give target-specific information
!      --  to formal verification tools.
! 
!      for Switches ("system.ads") use Target_Options.ALL_ADAFLAGS
!        & ("-gnatet=" & Project'Project_Dir & "/ada_target_properties");
!   end Compiler;
  
  end Runtime_Build;
--- 1,30 ----
  with "target_options.gpr";
+ --  with "bsp_build.gpr";
  
  project Runtime_Build is
! 
!    for Languages use ("Ada", "C", "Asm_Cpp");
!    for Runtime ("ada") use project'Project_Dir;
!    for Target use "arm-eabi";
!    for Library_Auto_Init use "False";
!    for Library_Name use "gnat";
!    for Library_Kind use "static";
!    for Library_Dir use "adalib";
!    for Object_Dir use "obj";
!    for Source_Dirs use ("gnat_user", "gnat", "gnarl_user", "gnarl", "bsp", "bsp/STM32F7xx_HAL_Driver/Src", "bsp/STM32F7xx_HAL_Driver/Inc", 
!                         "bsp/CMSIS/Include", "bsp/CMSIS/Device/ST/STM32F7xx/Include", "bsp/conf", 
!                         "bsp/BSP/STM32746G-Discovery", "bsp/BSP/Components/Common", "bsp/BSP/Components/ft5336", "bsp/BSP/Components/lis2mdl", 
!                         "bsp/BSP/Components/lsm6ds3", "bsp/BSP/Components/rk043fn48h", "bsp/BSP/Components/wm8994", "bsp/BSP/Components/n25q128a", 
!                         "bsp/BSP/Components/bmp280", "bsp/res", "bsp/wrapper", "bsp/tests", "bsp/hal_stm_lvgl/tft", "bsp/hal_stm_lvgl/touchpad", 
!                         "bsp/lvgl/src/lv_core", "bsp/lvgl/src/lv_draw", "bsp/lvgl/src/lv_font", "bsp/lvgl/src/lv_gpu", "bsp/lvgl/src/lv_hal", 
!                         "bsp/lvgl/src/lv_misc", "bsp/lvgl/src/lv_themes", "bsp/lvgl/src/lv_widgets");
! 
!    package Compiler renames Target_Options.Compiler;
! 
!    package Builder is
!       for Switches ("ada") use ("-j4", "-g");
!       for Switches ("C") use ("-j4", "-g");
!    end Builder;
  
  end Runtime_Build;
+ 
diff -c /usr/local/insa/GNAT-2021-dnu/2021/arm-elf/arm-eabi/lib/gnat/ravenscar-full-stm32f746disco/runtime.xml ./runtime.xml
*** /usr/local/insa/GNAT-2021-dnu/2021/arm-elf/arm-eabi/lib/gnat/ravenscar-full-stm32f746disco/runtime.xml	2025-04-03 17:37:05.350846064 +0200
--- ./runtime.xml	2021-11-17 09:57:08.000000000 +0100
***************
*** 7,13 ****
     Loader : Loaders := external("LOADER", "ROM");
  
     package Compiler is
!       Common_Required_Switches := ("-mlittle-endian", "-mfloat-abi=hard", "-mcpu=cortex-m7", "-mfpu=fpv5-sp-d16", "-mthumb");
  
        for Leading_Required_Switches ("Ada") use
           Compiler'Leading_Required_Switches ("Ada") &
--- 7,13 ----
     Loader : Loaders := external("LOADER", "ROM");
  
     package Compiler is
!       Common_Required_Switches := ("-mlittle-endian", "-mhard-float", "-mcpu=cortex-m7", "-mfpu=fpv5-sp-d16", "-mthumb");
  
        for Leading_Required_Switches ("Ada") use
           Compiler'Leading_Required_Switches ("Ada") &
***************
*** 29,35 ****
     package Linker is
        for Required_Switches use Linker'Required_Switches &
          ("-Wl,-L${RUNTIME_DIR(Ada)}/adalib",
!          "-nostartfiles", "-lc", "-lgnarl", "-lgnat", "-lgnarl",
           "-L${RUNTIME_DIR(ada)}/ld_user",
           "-L${RUNTIME_DIR(ada)}/ld") &
           Compiler.Common_Required_Switches;
--- 29,35 ----
     package Linker is
        for Required_Switches use Linker'Required_Switches &
          ("-Wl,-L${RUNTIME_DIR(Ada)}/adalib",
!          "-nostartfiles", "-lc", "-lgnat",
           "-L${RUNTIME_DIR(ada)}/ld_user",
           "-L${RUNTIME_DIR(ada)}/ld") &
           Compiler.Common_Required_Switches;
diff -c /usr/local/insa/GNAT-2021-dnu/2021/arm-elf/arm-eabi/lib/gnat/ravenscar-full-stm32f746disco/target_options.gpr ./target_options.gpr
*** /usr/local/insa/GNAT-2021-dnu/2021/arm-elf/arm-eabi/lib/gnat/ravenscar-full-stm32f746disco/target_options.gpr	2025-04-03 17:37:35.907929234 +0200
--- ./target_options.gpr	2025-07-02 14:07:12.790704553 +0200
***************
*** 8,48 ****
    type Build_Type is ("Production", "Debug", "Assert", "Gnatcov");
    Build : Build_Type := external ("BUILD", "Production");
  
!   --  COMFLAGS here is common flags (used for C and Ada).
!   COMFLAGS := ("-fcallgraph-info=su,da",
          "-ffunction-sections",
          "-fdata-sections");
-   COMGNARLFLAGS := ("");
  
!   --  Per language flags (COMFLAGS will be added later)
!   ADAFLAGS := ("-gnatg", "-nostdinc", "-fno-delete-null-pointer-checks");
!   ASMFLAGS := ("");
!   CFLAGS := ("-DIN_RTS",
!         "-Dinhibit_libc", "-Werror", "-Wall");
!   case Build is
!      when "Production" =>
!         --  Optimize
!         COMFLAGS := COMFLAGS & ("-O2");
!         ADAFLAGS := ADAFLAGS & ("-gnatp", "-gnatn2");
!      when "Debug" =>
!         --  Disable optimization and add debug symbols
!         COMFLAGS := COMFLAGS & ("-O0", "-g");
!         ASMFLAGS := ASMFLAGS & ("-g");
!      when "Assert" =>
!         --  Possibly enable assertions. This might use too much memory on
!         --  some systems or could be too slow.
!         COMFLAGS := COMFLAGS & ("-O");
!         ADAFLAGS := ADAFLAGS & ("-gnata");
!      when "Gnatcov" =>
!         --  For coverage
!         COMFLAGS := COMFLAGS & ("-O0", "-g", "-fdump-scos",
!                                 "-fpreserve-control-flow");
!   end case;
! 
!   --  Concatenate with common flags
!   ALL_ADAFLAGS := ADAFLAGS & COMFLAGS;
!   ALL_CFLAGS := CFLAGS & COMFLAGS;
!   GNARL_ADAFLAGS := ALL_ADAFLAGS & COMGNARLFLAGS;
!   GNARL_CFLAGS := ALL_CFLAGS & COMGNARLFLAGS;
  
  end Target_Options;
--- 8,77 ----
    type Build_Type is ("Production", "Debug", "Assert", "Gnatcov");
    Build : Build_Type := external ("BUILD", "Production");
  
!   package Naming is
!      for Spec_Suffix ("Asm_CPP") use ".inc";
!   end Naming;
! 
!   package Compiler is
!      --  COMFLAGS here is common flags (used for C and Ada).
!      COMFLAGS := ("-fcallgraph-info=su,da",
          "-ffunction-sections",
          "-fdata-sections");
  
!      --  Per language flags (COMFLAGS will be added later)
!      ADAFLAGS := ("-gnatg", "-nostdinc", "-fno-delete-null-pointer-checks", "-gnatwa", "-gnatwR", "-gnaty-z");
!      ASMFLAGS := ("");
!       CFLAGS := ("-DIN_RTS","-Dinhibit_libc", "-Werror", "-Wall", "-DSTM32F746xx",
!                  "-DUSE_HAL_DRIVER", "-DLV_EX_CONF_INCLUDE_SIMPLE", "-DDEBUG",
!                  "-DSTM32F746G_DISCO", "-DSTM32F746NGHx", "-DSTM32F7");
!      case Build is
!         when "Production" =>
! 	   --  Optimize
!            COMFLAGS := COMFLAGS & ("-O2");
!            ADAFLAGS := ADAFLAGS & ("-gnatp", "-gnatn2");
!         when "Debug" =>
! 	   --  Disable optimization and add debug symbols
!            COMFLAGS := COMFLAGS & ("-O0", "-dwarf", "-g3");
!            ASMFLAGS := ASMFLAGS & ("-g3");
! 	when "Assert" =>
! 	   --  Possibly enable assertions. This might use too much memory on
! 	   --  some systems or could be too slow.
!            COMFLAGS := COMFLAGS & ("-O");
! 	   ADAFLAGS := ADAFLAGS & ("-gnata");
!         when "Gnatcov" =>
! 	   --  For coverage
!            COMFLAGS := COMFLAGS & ("-O0", "-g", "-fdump-scos",
!                                "-fpreserve-control-flow");
!      end case;
! 
!      --  Concatenate with common flags
!      ALL_ADAFLAGS := ADAFLAGS & COMFLAGS;
!      ALL_CFLAGS := CFLAGS & COMFLAGS;
! 
!      for Switches ("C") use ALL_CFLAGS;
!      for Switches ("Ada") use ALL_ADAFLAGS;
!      for Switches ("Asm_Cpp") use ASMFLAGS;
! 
!      --  Some runtime files need to be compiled with debug info, so that gdb
!      --  is not blind.
!      for Switches ("s-traceb.adb") use ALL_ADAFLAGS & ("-g3")
!        & ("-fno-optimize-sibling-calls", "-fno-inline-functions-called-once");
!      for Switches ("s-tasdeb.adb") use ALL_ADAFLAGS & ("-g3", "-O0");
!      for Switches ("a-except.adb") use ALL_ADAFLAGS
!        & ("-g", "-O1", "-fno-inline", "-fno-toplevel-reorder");
!      for Switches ("s-excdeb.adb") use ALL_ADAFLAGS & ("-g3", "-O0");
!      for Switches ("s-assert.adb") use ALL_ADAFLAGS & ("-g3");
!      for Switches ("a-tags.adb") use ALL_ADAFLAGS & ("-g3");
!      for Switches ("raise-gcc.c") use ALL_CFLAGS & ("-fexceptions");
! 
!      --  Don't inline System.Machine_Reset otherwise we can loose our common
!      --  exit system.
! 
!      for Switches ("s-macres.adb") use ALL_ADAFLAGS & ("-fno-inline");
! 
!      for Switches ("system.ads") use ALL_ADAFLAGS &
!        ("-gnatet=" & Project'Object_Dir & "/../ada_target_properties");
! 
!   end Compiler;
  
  end Target_Options;
